src/gurt/telemetry.c

```diff
@@ -1075,6 +1075,49 @@ d_tm_compute_histogram(struct d_tm_node_t *node, uint64_t value)
 	return rc;
 }
 
+/**
+ * Set the given counter to the specified \a value
+ *
+ * \param[in,out]	metric	Pointer to the metric
+ * \param[in]		value	Increments the counter by this \a value
+ */
+void
+d_tm_set_counter(struct d_tm_node_t *metric, uint64_t value)
+{
+	if (unlikely(d_tm_shmem_root == NULL || metric == NULL))
+		return;
+
+	if (unlikely(metric->dtn_type != D_TM_COUNTER)) {
+		D_ERROR("Failed to set counter [%s] on item not a "
+			"counter.\n", metric->dtn_name);
+		return;
+	}
+
+	metric->dtn_metric->dtm_data.value = value;
```

Comment:
@kjacque I know that this is not compatible with other d_tm functions, but i would like to move towards this path where we always requires telemetry counters to be explicitly pre-created and discard errors.

Reply:
Makes sense to me. In the event that some telemetry event fails, I don't know that there's anything to do anyway. In practice I think we were only checking the error codes in tests.